src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -0,0 +1,184 @@
+ * following-sibling::METHOD_DEF[.//LITERAL_PUBLIC]]"/&gt; &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public void method1() { }
+ *  private void method2() { } // violation
+ *  public void method3() { }
+ *  private void method4() { } // violation
+ *  public void method5() { }
+ *  private void method6() { } // ok
+ * }
+ * </pre>
+ * <p>
+ * To fail with violation if there are any parametrized constructors
+ * </p>
+ * <pre>
+ * &lt;module name="MatchXpath"&gt; &lt;property name="query"
+ * value="//CTOR_DEF[count(./PARAMETERS/node()) &gt; 0]"/&gt; &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public Test(Object c) { } // violation
+ *  public Test(int a, HashMap&lt;String, Integer&gt; b) { } // violation
+ *  public Test() { } // ok
+ * }
+ * </pre>
+ * <p>
+ * To fail with violation if method name is 'test' or 'foo'
+ * </p>
+ * <pre>
+ * &lt;module name="MatchXpath"&gt; &lt;property name="query"
+ * value="//METHOD_DEF[./IDENT[@text='test' or @text='foo']]"/&gt; &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public void test() {} // violation
+ *  public void getName() {} // ok
+ *  public void foo() {} // violation
+ *  public void sayHello() {} // ok
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code matchxpath.match}
+ * </li>
+ * </ul>
+ *
+ * @since 8.38
+ */
+@StatelessCheck
+public class MatchXpathCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text provided by user.
+     */
+    public static final String MSG_KEY = "matchxpath.match";
+
+    /** Xpath query. */
+    private String query = "";
+
+    /**
+     * Setter to xpath query.
+     *
+     * @param query xpath query.
+     */
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        if (!query.isEmpty()) {
+            final List<DetailAST> matchingNodes = findMatchingNodesByXpathQuery(rootAST);
+            matchingNodes.forEach(node -> log(node, MSG_KEY));
+        }
+    }
+
+    private List<DetailAST> findMatchingNodesByXpathQuery(DetailAST rootAST) {
+        final XPathEvaluator xpathEvaluator = new XPathEvaluator(Configuration.newConfiguration());
+        try {
+            final RootNode rootNode = new RootNode(rootAST);
+            final XPathExpression xpathExpression = xpathEvaluator.createExpression(query);
```

Comment:
```
        final XPathEvaluator xpathEvaluator = new XPathEvaluator(Configuration.newConfiguration());
        final XPathExpression xpathExpression = xpathEvaluator.createExpression(query);
```

can we make this execution to be done ones for whole Checkstyle execution ?
for example in "setQuery".

Reply:
done